lib/msf/core/post/file_stat.rb

```diff
@@ -0,0 +1,169 @@
+# -*- coding: binary -*-
+#
+module Msf::Post::FileStat
+
+class Stat
```

Comment:
How about inheriting from `Rex::Post::FileStat` instead? You'd need to update the `#initialize` method but then everything else should be inherited. You should just need to set the `stathash` attribute based on the keys you're generating here. It would also address a lot of the inconsistencies with the Meterpreter version since that also inherits from `Rex::Post::FileStat`.

This should also probably be updated to:

```suggestion
module Msf::Post

class FileStat
```

So the class is `FileStat` just like in Rex and it exists under the File namespace.

Reply:
sure! that will be much better because both of them will be inherit from `Rex::Post::FileStat`